src/detect.h

```diff
@@ -587,6 +588,11 @@ typedef struct DetectEngineCtx_ {
     ThresholdCtx ths_ctx;
 
     uint16_t mpm_matcher; /**< mpm matcher this ctx uses */
+    uint16_t spm_matcher; /**< spm matcher this ctx uses */
+
+    /* spm thread context prototype, built as spm matchers are constructed and
+     * later cloned for each thread. */
+    SpmThreadCtx *spm_thread_ctx;
```

Comment:
I like the idea of using a different type, which will make it harder to accidentally use the wrong structure at scan time. Will add an SpmGlobalCtx structure.